dynamiqs/_checks.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import jax.numpy as jnp
+from jax import Array
+
+_is_perfect_square = lambda n: int(n**0.5) ** 2 == n
+
+
+def has_shape(x: Array, shape: str) -> bool:
+    cases = {
```

Comment:
Missing (maybe intentional):
 - (N, 1, n) for batched bras
 - (..., n^2, n^2) for vectorized Liouvillians
 - (..., 1, n^2) for daggered vectorized density matrices

Reply:
Yes it was intentional, I just added checks that we actually use in the codebase. We can add new checks as we go along when we need them.